src/globals/scss/_layer.scss

```diff
@@ -35,10 +35,12 @@ $layer-shadows: (
 
 // Layer mixin
 @mixin layer($layer) {
-  @if map-has-key($layer-shadows, $layer) {
-    box-shadow: #{map-get($layer-shadows, $layer)};
-  }
-  @else {
-    @warn '#{$layer} is not a valid layer.';
+  @if ($css--use-layer) {
```

Comment:
```sh
Error in plugin 'sass'
Message:
    src/globals/scss/_layer.scss
Error: Undefined variable: "$css--use-layer".
        on line 38 of src/globals/scss/_layer.scss
>>   @if ($css--use-layer) {
```

This mixin forces the user to always have the variable declared, otherwise they get an undefined variable error.
You can match the [conditional logic we have for other global variables](https://github.com/carbon-design-system/carbon-components/pull/76/files#diff-02222be8ee9618c8b4dbd6a9a584d102R29).

```scss
@if global-variable-exists('css--use-layer') == false or $css--use-layer == true {
  @include layer($layer); // layer mixin can remove if/else statement and warning and just keep box-shadow css
}
```

__Here's how it works:__ the `global-variable-exists('css--use-layer')` SCSS function checks ["whether a variable with the given name exists in the global scope (at the top level of the file)."](http://sass-lang.com/documentation/Sass/Script/Functions.html#global_variable_exists-instance_method). 

- __If the variable does not exist__: the layer mixin will get invoked.
- __If the variable exists__: then we check the boolean value of the variable.
  - __If variable is true__: invoke the mixin
  - __If variable is false__: do nothing.

__Basically:__ The mixin always gets invoked unless the user explicitly sets the global variable to `false`